File: Sirius/Models/Invoice.cs
using System;
using System.Collections.Generic;

namespace Sirius.Models
{
    /// <summary>
    /// Накладные
    /// </summary>
    public class Invoice
    {
        /// <summary>
        /// Уникальный идентификатор
        /// </summary>
        public Guid Id { get; set; }

        /// <summary>
        /// Заголовок
        /// </summary>
        public string Name { get;set; }

        /// <summary>
        /// Идентификатор поставщика
        /// </summary>
        public Guid VendorId { get; set; }

        /// <summary>
        /// Поставщик
        /// </summary>
        public Vendor Vendor { get; set; }

        /// <summary>
        /// Идентификатор пользователя-автора
        /// </summary>
        public Guid UserId { get; set; }

        /// <summary>
        /// Пользователь-автор
        /// </summary>
        public User User { get; set; }

        /// <summary>
        /// Список регистров
        /// </summary>
        public IEnumerable<Register> Registers { get; set; }

        /// <summary>
        /// Признак проведённой накладной
        /// </summary>
        public bool IsFixed { get; set; }

        /// <summary>
        /// Признак временной накладной
        /// </summary>
        public bool IsTemporary { get; set; }

        /// <summary>
        /// Дата создания 
        /// </summary>
        public DateTime CreateDate { get; set; }

        /// <summary>
        /// Идентификатор типа накладной
        /// </summary>
        public Guid TypeId { get; set; }

        /// <summary>
        /// Комментарий
        /// </summary>
        public string Comment { get; set; }

        /// <summary>
        /// Фактор
        /// </summary>
        public double Factor { get; set; }
    }
}

File: Sirius/ClientApp/app/components/_models/invoice.ts
import { Vendor, User, Register } from ".";

export class Invoice {
    id: string;
    name: string;
    userId: string;
    userFullName: string;
    vendorId: string;
    vendorName: string;
    registers: Register[];
    date: string;
    isFixed: boolean;
    isTemporary: boolean;
    typeId: string;
    factor: number;
    comment: string;
}

export class InvoiceListItem {
    id: string;
    name: string;
    userFullName: string;
    date: string;
    isFixed: boolean;
    isTemporary: boolean;
}

export class InvoiceUpdate {
    id: string;
    name: string;
    vendorid: string;
    userid: string;
    isrecorderd: boolean;
    istemporary: boolean;
    date: string;
}

export class InvoiceType {
    id: string;
    name: string;
    factor: number;
    alias: string;
}